### ISSUE TITLE
\Magento\Config\Model\Config\Source\Email\Template::toOptionArray throws error when setPath() is not called first

### ISSUE THREAD

**Issue by willwright** [magento/magento2#29315]
### Preconditions (*)

1. dev-2.4-develop

### Steps to reproduce (*)

1. Create a custom module, plugin, or Observer
2. Inject a concrete instance of \Magento\Framework\ObjectManagerInterface ($this->_objectManager) through the constructor
3. Execute the following:
```
$sourceModelObj = $this->_objectManager->create('\Magento\Config\Model\Config\Source\Email\Template');
$optionsArray = $sourceModelObj->toOptionArray();
```
4. Notice unhandled exception is written to the screen

### Expected result (*)

1. **Should not** throw an error
2. **Should** return default collection options

### Actual result (*)

```
Exception #0 (UnexpectedValueException): Email template '' is not defined.
<pre>#1 Magento\Email\Model\Template\Config->getTemplateLabel('') called at [vendor/magento/module-config/Model/Config/Source/Email/Template.php:64]
#2 Magento\Config\Model\Config\Source\Email\Template->toOptionArray() called at [app/code/Magesmith/ScopeExplorer/Block/Scopes.php:172]
#3 Magesmith\ScopeExplorer\Block\Scopes->getSourceModelValue('catalog_producta...') called at [app/code/Magesmith/ScopeExplorer/Block/Scopes.php:135]
#4 Magesmith\ScopeExplorer\Block\Scopes->getWebsiteScopeValue(&Magento\Store\Model\Website\Interceptor#00000000476d138d000000000886eb4d#) called at [app/code/Magesmith/ScopeExplorer/Block/Scopes.php:97]
#5 Magesmith\ScopeExplorer\Block\Scopes->hasScopeValue() called at [app/code/Magesmith/ScopeExplorer/view/adminhtml/templates/scopes.phtml:7]
#6 include('/mnt/f/www/local...') called at [vendor/magento/framework/View/TemplateEngine/Php.php:59]
#7 Magento\Framework\View\TemplateEngine\Php->render(&Magesmith\ScopeExplorer\Block\Scopes#00000000476d1f2e000000000886eb4d#, '/mnt/f/www/local...', array('csp' => &Magento\Csp\Helper\InlineUtil#00000000476d1d31000000000886eb4d#)) called at [vendor/magento/framework/Interception/Interceptor.php:58]
```

---
Please provide [Severity](https://devdocs.magento.com/guides/v2.3/contributor-guide/contributing.html#backlog) assessment for the Issue as Reporter. This information will help during Confirmation and Issue triage processes.

- [ ] Severity: **S0** _- Affects critical data or functionality and leaves users without workaround._
- [ ] Severity: **S1** _- Affects critical data or functionality and forces users to employ a workaround._
- [ ] Severity: **S2** _- Affects non-critical data or functionality and forces users to employ a workaround._
- [x ] Severity: **S3** _- Affects non-critical data or functionality and does not force users to employ a workaround._
- [ ] Severity: **S4** _- Affects aesthetics, professional look and feel, “quality” or “usability”._

**Comment by magento-engcom-team** [comment]
@engcom-Alfa Thank you for verifying the issue.

Unfortunately, not enough information was provided to acknowledge ticket. Please consider adding the following:
- [ ] Add `"Component: "` label(s) to this ticket based on verification result. If uncertain, you may follow the best guess

Once all required information is added, please add label `"Issue: Confirmed"` again.
Thanks!

**Comment by magento-engcom-team** [comment]
:white_check_mark: Confirmed by @engcom-Alfa 
 Thank you for verifying the issue. Based on the provided information internal tickets `MC-36860` were created

**Issue Available**: @engcom-Alfa, _You will be automatically unassigned. Contributors/Maintainers can claim this issue to continue. To reclaim and continue work, reassign the ticket to yourself._

**Comment by magento-engcom-team** [comment]
Hi @willwright. Thank you for your report. 
The issue has been fixed in magento/magento2#29316 by @willwright in 2.4-develop branch
Related commit(s):
 - [e786d514ef72699bdedada6444665d882dd015c3](https://github.com/magento/magento2/commit/e786d514ef72699bdedada6444665d882dd015c3)
 - [e0972bf2ac019a1b4645f2aaa7ac2e6016859fe5](https://github.com/magento/magento2/commit/e0972bf2ac019a1b4645f2aaa7ac2e6016859fe5)

The fix will be available with the upcoming 2.4.2 release.
